spec/output/generator_spec_elixir/test.x/nested_enum.ex

```diff
@@ -0,0 +1,56 @@
+defmodule MyXDR.nestedEnum do
+  @moduledoc """
+  Automatically generated by xdrgen
+  DO NOT EDIT or your changes may be overwritten
+
+  Target implementation: elixir_xdr at https://hex.pm/packages/elixir_xdr
+
+  Representation of Stellar `nestedEnum` type.
+  """
+
+  @behaviour XDR.Declaration
+
+  @declarations [
+    BLAH_1: 0,
+    BLAH_2: 1
+  ]
+
+  @enum_spec %XDR.Enum{declarations: @declarations, identifier: nil}
+
+  @type t :: %__MODULE__{identifier: atom()}
+
+  defstruct [:identifier]
+
+  @spec new(type :: atom()) :: t()
+  def new(type \\ :BLAH_1), do: %__MODULE__{identifier: type}
+
+  @impl true
+  def encode_xdr(%__MODULE__{identifier: type}), do:
+    @declarations
+    |> XDR.Enum.new(type)
+    |> XDR.Enum.encode_xdr()
```

Comment:
IMO the notation `do ... end` looks better for these pipelines  in the encode/decode functions